src/stories/components/custom-multi-select-2/custom-multi-select-2.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useRef, useState } from 'react';
+import styled from '@emotion/styled';
+import { SelectChevronDown } from '../svg/select-chevron-down';
+import { Checkbox, ListItemText, MenuItem, Typography } from '@mui/material';
+import CheckBoxOutlined from '@mui/icons-material/CheckBoxOutlined';
+import './custom-multi-select-2.scss';
+import useOutsideClick from '../../../core/utils/use-outside-click';
+
+interface CustomMultiSelect2Props {
+  label: string,
+  items: string[],
+  withAll?: boolean,
+  onChange?: (items: string[]) => void,
+  initialActiveItems?: string[],
+}
+
+export const CustomMultiSelect2 = ({ withAll = true, initialActiveItems = [], ...props }: CustomMultiSelect2Props) => {
+  const [visible, setVisible] = useState(false);
+  const [activeItems, setActiveItems] = React.useState<string[]>(initialActiveItems);
+
+  const refOutsideClick = useRef();
+
+  useOutsideClick(refOutsideClick, () => {
+    visible && setVisible(false);
+  });
+
+  const toggleItem = (item: string) => {
+    const pos = activeItems.indexOf(item);
+    if (pos > -1) {
+      const temp = [...activeItems];
+      temp.splice(pos, 1);
+      setActiveItems(temp);
+    } else {
+      const temp = [...activeItems];
+      temp.push(item);
+      setActiveItems(temp);
+    }
+  };
+
+  const toggleVisible = () => setVisible(!visible);
+
+  const toggleAll = () => {
+    if (activeItems.length === props.items.length) {
+      setActiveItems([]);
+      props.onChange && props.onChange([]);
```

Comment:
I think you can destructuring the props that you are going to need, then you can avoid write props.something, of course its not necessary, but for me look much easy to read

Reply:
TBH I don't agree. Having the properties inside the props object helps with clarity, it's easier to identify which variables are local and which come from the father component. Personally I only use "spread" when assigning default values to it, but it was good to get your point of view from it.